internal/toolexec/aspect/oncompile.go

```diff
@@ -62,10 +62,56 @@ var weavingSpecialCase = []specialCase{
 	{path: "github.com/DataDog/go-tuf/client", prefix: false, behavior: neverWeave},
 }
 
-func (w Weaver) OnCompile(cmd *proxy.CompileCommand) error {
+func (w Weaver) OnCompile(cmd *proxy.CompileCommand) (result error) {
```

Comment:
⚠️ I'm going to do a refactor of this monstrosity tomorrow before we can merge this.